ci/test_python.sh

```diff
@@ -127,13 +127,6 @@ python -m nx_cugraph.scripts.print_tree --dispatch-name --plc --incomplete --dif
 python -m nx_cugraph.scripts.print_table
 popd
 
-rapids-logger "ensure nx-cugraph autogenerated files are up to date"
-pushd python/nx-cugraph
-make || true
-git diff --exit-code .
-git checkout .
-popd
-
```

Comment:
This check can still serve a purpose. The README is not the only thing that gets automatically updated. `make plugin-info` is used to update `_nx_cugraph/__init__.py`, and this emphatically should _not_ be a pre-commit in a dev environment, because a dev environment may be using a different version of networkx than CI uses. Running `make` also performs linting, which is a nice-to-check.

Reply:
Thanks for pointing that out.  I was able to add an additional hook to do the plugin-info update too.  I also agree about the need to isolate a known version of NetworkX to use for these codegen steps from the dev env, so I added a specific NetworkX dependency to the hook environment, which should guarantee consistency.  Let me know what you think, and I can still revert this if necessary since the only requirements for this PR are the README updates.

Reply:
But, I guess we still don't have the same level of linting.  I did update `_nx_cugraph/core.py` to generate black-compatible strings, which now pass `black`, but this is still not running `ruff`, `isort`, etc.

> Any appetite for updating linting in pre-commit to better match what we have in nx-cugraph?

I like the idea of the nx-cugraph level of linting though.  I can either open an issue for 24.06 to update the cugraph pre-commit to match, or I can add `make lint` to the `test_python.sh` script (at least until cugraph catches up).  What do you think?

Reply:
We discussed this offline and decided to file [this issue](https://github.com/rapidsai/cugraph/issues/4234) rather than change this PR.